experimental/SymmetricIntersections/Representations.jl

```diff
@@ -0,0 +1,1465 @@
+
+# We compute all consistuents of chi of degree t
+@doc Markdown.doc"""
+    constituents(chi::Oscar.GAPGroupClassFunction, t::Int)
+                                     -> Vector{Oscar.GAPGroupClassFunction)
+
+Given a character `chi` and an integer `t`, return all the consistuents of
+`chi` of degree `t`.
+"""
+function constituents(chi::Oscar.GAPGroupClassFunction, t::Int)
+  @req is_character(chi) "chi should be a character"
+  @req t >= 0 "t must be non negative"
+  
+  # sort of fallback when t = degree(chi)
+  if t == 0
+    return [zero(chi)]
+  end
+  
+  # constituents of degree t in chi corresponds bijectively to their
+  # complement which are of degree degree(chi)-t. So to do the smallest
+  # computations, we always reduce to the case t <= floor(degree(chi)/2)
+  if t > div(Int(degree(chi)), 2)
+    els = constituents(chi, Int(degree(chi)) - t)
+    return [chi-nu for nu in els]
+  end
+  
+  cd = character_decomposition(chi)
+  L = [c[2] for c in cd]
+  ubs = [c[1] for c in cd]
+  
+  function _deg(x::Oscar.GAPGroupClassFunction)
+    return ZZ(degree(x))
+  end
+  
+  # _deg(L) represents the degree of all irreducible constituents of chi,
+  # and ubs keeps track of their multiplicities in chi. So that we return
+  # all the t-elevations of (L, _deg) where we impose some upper bound
+  # on their of time we can take each index (otherwise the resulting
+  # character won't be a constituent of chi
+  el = elevator(L, _deg, t, ubs = ubs)
+  (number_of_elevations(el) == 0) && return Oscar.GAPGroupClassFunction[]
+  return Oscar.GAPGroupClassFunction[sum(L[l]) for l in el]
+end
+
+@doc Markdown.doc"""
+    all_characters(RR::RepRing, t::Int) -> Vector{Oscar.GAPGroupClassFunction}
+
+Given a representation ring `RR` associated to a finite group `E` and an integer
+`t`, return all characters of `E` of degree `t`.
+"""
+function all_characters(RR::RepRing, t::Int)
+  chis = irreducible_characters_underlying_group(RR)
+  el = elevator(chis, x -> ZZ(degree(x)), t)
+  return Oscar.GAPGroupClassFunction[sum(chis[l]) for l in el]
+end
+
+@doc Markdown.doc"""
+    determinant(chi::Oscar.GAPGroupClassFunction) -> Oscar.GAPGroupClassFunction
+
+Given a character `chi` of degree `n`, return its determinant character,
+which corresponds to the `n`-th antisymmetric parts of `chi`.
+"""
+determinant(chi::Oscar.GAPGroupClassFunction) = exterior_power(chi, Int(degree(chi)))
+
+@doc Markdown.doc"""
+    center_of_character(chi::Oscar.GAPGroupClassFunction) -> GAPGroup, GAPGroupHomomorphism
+
+Return the center of the character `chi` which corresponds to the subgroup
+of the underlying group `E` of `chi` consisting on the elements `e` of `E`
+such that $chi(e)$ is a primitive root of unity times the degree of `chi`.
+"""
+function center_of_character(chi::Oscar.GAPGroupClassFunction)
+  E = chi.table.GAPGroup
+  _H = GG.CenterOfCharacter(chi.values)::GAP.GAP_jll.GapObj
```

Comment:
Use `GAP.GapObj` not `GAP.GAP_jll.GapObj`.
(The latter is no longer supported in GAP.jl  0.9.3.)